morphct/system.py

```diff
@@ -0,0 +1,247 @@
+        dimer_data = dimer_homolumo(
+            self.qcc_pairs, self.chromophores, d_filename
+        )
+        t2 = time.perf_counter()
+        print(f"Finished in {t2-t1:.2f} s. Output written to {d_filename}.")
+
+    def set_energies(self, s_filename, d_filename, dcut=None):
+        """
+
+        Parameters
+        ----------
+
+        """
+        if self.qcc_pairs is None:
+            if dcut is None:
+                dcut = min(self.snap.configuration.box[:3]/2)
+            self.qcc_pairs = set_neighbors_voronoi(
+                self.chromophores, self.snap, self.conversion_dict, d_cut=dcut
+            )
+        set_energyvalues(self.chromophores, s_filename, d_filename)
+        print("Energies set.")
+
+    def run_kmc(
+        self,
+        lifetimes,
+        temp,
+        n_holes=0,
+        n_elec=0,
+        seed=42,
+        combine=True,
+        carrier_kwargs={},
+        verbose=0
+    ):
+        kmc_dir = os.path.join(self.outpath, "kmc")
+        if not os.path.exists(kmc_dir):
+            os.makedirs(kmc_dir)
+        data = run_kmc(
+            lifetimes,
+            kmc_dir,
+            self.chromophores,
+            self.snap,
+            temp,
+            n_holes=n_holes,
+            n_elec=n_elec,
+            seed=seed,
+            combine=combine,
+            carrier_kwargs=carrier_kwargs,
+            verbose=verbose
+        )
+
+        kmc_analyze.main(data, temp, self.chromophores, self.snap, kmc_dir)
+
+    def visualize_qcc_input(self, i, single=True):
+        """Visualize the input for pyscf using mbuild.
+
+        Parameters
+        ----------
+        qcc_input : str
+            Input string to visualize
+        """
+        import mbuild as mb
+
+        if single:
+            qcc_input = self.chromophores[i].qcc_input
+        else:
+            qcc_input = self.qcc_pairs[i][1]
+        comp = mb.Compound()
+        for line in qcc_input.split(";")[:-1]:
+            atom, x, y, z = line.split()
+            xyz = np.array([x,y,z], dtype=float)
+            # Angstrom -> nm
+            xyz /= 10
+            comp.add(mb.Particle(name=atom, pos=xyz))
+        comp.visualize().show()
+
+    def visualize_system(self):
+        """Visualize the system snapshot."""
+        if self._comp is None:
+            self._comp = self._make_comp()
+        self._comp.visualize().show()
+
+    def visualize_chromophores(self):
+        """Visualize the system snapshot."""
+        import mbuild as mb
+
+        if self._comp is None:
+            self._comp = self._make_comp()
+        ccomp = mb.clone(self._comp)
+        for i,p in enumerate(ccomp):
+            if self._ainds:
+                if i in np.hstack(self._ainds):
+                    p.name = "Br"
+            if self._dinds:
+                if i in np.hstack(self._dinds):
+                    p.name = "I"
+        ccomp.visualize().show()
+
+    def _make_comp(self):
```

Comment:
yeah good point 😅

Reply:
OK, I think the reason I did this is because I don't want to install mbuild from source... when they release the next conda build, I'll update it. Also this function doesn't set the box because I found that slowed things down.